config/management/operational/src/network_checker.rs

```diff
@@ -2,89 +2,83 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use crate::json_rpc::JsonRpcClientWrapper;
-use diem_config::config::RoleType;
+use diem_config::{
+    config::{RoleType, HANDSHAKE_VERSION},
+    network_id::{NetworkContext, NetworkId},
+};
+use diem_crypto::x25519;
 use diem_management::error::Error;
 use diem_network_address_encryption::Encryptor;
 use diem_secure_storage::{InMemoryStorage, Storage};
-use diem_types::network_address::{
-    encrypted::{Key, KeyVersion, KEY_LEN},
-    NetworkAddress,
+use diem_types::{
+    account_address,
+    chain_id::ChainId,
+    network_address::{
+        encrypted::{Key, KeyVersion, KEY_LEN},
+        NetworkAddress,
+    },
+    PeerId,
 };
 use fallible::copy_from_slice::copy_slice_to_vec;
-use std::{
-    io::{Read, Write},
-    net::{TcpStream, ToSocketAddrs},
-    time::Duration,
+use netcore::transport::tcp::{resolve_and_connect, TcpSocket};
+use network::{
+    noise::{HandshakeAuthMode, NoiseUpgrader},
+    protocols::wire::handshake::v1::SupportedProtocols,
+    transport::{upgrade_outbound, UpgradeContext, SUPPORTED_MESSAGING_PROTOCOL},
+    ProtocolId,
 };
+use std::{collections::BTreeMap, sync::Arc};
 use structopt::StructOpt;
+use tokio::{runtime::Runtime, time::Duration};
+
+const DEFAULT_TIMEOUT: Duration = Duration::from_secs(5);
 
 #[derive(Debug, StructOpt)]
 pub struct CheckEndpoint {
+    /// `NetworkAddress` of remote server interface
     #[structopt(long)]
     address: NetworkAddress,
+    /// `ChainId` of remote server
+    #[structopt(long)]
+    chain_id: ChainId,
+    /// `NetworkId` of remote server interface
+    #[structopt(long)]
+    network_id: NetworkId,
+    #[structopt(long, parse(try_from_str = parse_private_key_hex))]
```

Comment:
nit: maybe add a comment to this so it's obvious to those who don't know what the private_key is for?